vizro-core/src/vizro/models/_components/aggrid.py

```diff
@@ -0,0 +1,121 @@
+
+    # Convenience wrapper/syntactic sugar.
+    def __call__(self, **kwargs):
+        kwargs.setdefault("data_frame", data_manager._get_component_data(self.id))
+        return self.figure(**kwargs)
+
+    # Convenience wrapper/syntactic sugar.
+    def __getitem__(self, arg_name: str):
+        # See table implementation for more details.
+        if arg_name == "type":
+            return self.type
+        return self.figure[arg_name]
+
+    # Interaction methods
+    def _get_figure_interaction_input(self) -> Dict[str, State]:
+        """Required properties when using pre-defined `filter_interaction`."""
+        return {
+            "cellClicked": State(component_id=self._callable_object_id, component_property="cellClicked"),
+            "modelID": State(component_id=self.id, component_property="id"),  # required, to determine triggered model
+        }
+
+    def _filter_interaction(
+        self, data_frame: pd.DataFrame, target: str, ctd_filter_interaction: Dict[str, CallbackTriggerDict]
+    ) -> pd.DataFrame:
+        """Function to be carried out for pre-defined `filter_interaction`."""
+        ctd_cellClicked = ctd_filter_interaction["cellClicked"]
+        if not ctd_cellClicked["value"]:
+            return data_frame
+
+        # ctd_active_cell["id"] represents the underlying table id, so we need to fetch its parent Vizro Table actions.
+        source_table_actions = _get_component_actions(_get_parent_vizro_model(ctd_cellClicked["id"]))
+
+        for action in source_table_actions:
+            if action.function._function.__name__ != "filter_interaction" or target not in action.function["targets"]:
+                continue
+            column = ctd_cellClicked["value"]["colId"]
+            clicked_data = ctd_cellClicked["value"]["value"]
+            data_frame = data_frame[data_frame[column].isin([clicked_data])]
+
+        return data_frame
+
+    @_log_call
+    def pre_build(self):
+        if self.actions:
+            kwargs = self.figure._arguments.copy()
+
+            # taken from table implementation - see there for details
+            kwargs["data_frame"] = pd.DataFrame()
+            underlying_aggrid_object = self.figure._function(**kwargs)
+
+            if not hasattr(underlying_aggrid_object, "id"):
+                raise ValueError(
+                    "Underlying `AGGrid` callable has no attribute 'id'. To enable actions triggered by the `AGGrid`"
+                    " a valid 'id' has to be provided to the `AGGrid` callable."
+                )
+
+            self._callable_object_id = underlying_aggrid_object.id
+
+    def build(self):
+        return dcc.Loading(
+            html.Div(
+                [
+                    html.H3(self.title, className="table-title") if self.title else None,
+                    html.Div(
+                        dash_table.DataTable(**({"id": self._callable_object_id} if self.actions else {})), id=self.id
```

Comment:
I know this doesn't really do anything because it gets overwritten on page load, but it's confusing using `DataTable` here rather than `AGGrid`.

Reply:
Going to see if we can remove this, but if I just exchange this for the `dag.AgGrid` the AG Grid doesn't render properly anymore. Going to investigate why

Reply:
If you go with this solution: https://github.com/mckinsey/vizro/pull/289#discussion_r1497465481, does it work then? So just removing the `if self.actions:` line from `pre_build` method, and in the build method set:
`dash_table.DataTable(id=self._callable_object_id), id=self.id` line for Table, and
`dag.AgGrid(id=self._callable_object_id), id=self.id` line for AgGrid.

Reply:
I just tested it with `dag.AgGrid(id=self._callable_object_id), id=self.id` approach, and it seems like backend properly returns data, but somehow it's not rendered on the UI.

Reply:
@maxschulz-COL 
Everything seems to work as expected when "pagination=True" is defined 🤔.

After I figured out the following configuration works:
`dash_ag_grid(data_frame=pd.DataFrame(), id=self._callable_object_id)()`

I started by removing the default attributes and found that the minimum configuration required to make it work was:
`dag.AgGrid(id=self._callable_object_id, **{"dashGridOptions": {"pagination": True}}),`

Reply:
Now I pulled the latest PR changes (in this commit `"pagination"=True` is removed from the `dash_ag_grid` default configuration) and it starts to work even with the `dag.AgGrid(id=self._callable_object_id)` configuration. 

It seems like it doesn't work when the object with the same ID initially doesn't have pagination included, and after the on_page_load mechanism it has. Somehow it makes sense to me now since `"pagination"=True` probably creates some mechanism per component (binds it with the component ID). If the pagination is additionally turned on for the already existing component then this backend mechanism isn't created, so it results with the empty data on the screen.

Reply:
Yes, it does make sense - what a funny coincidence. Well since we do not want pagination anymore, all good then :)

Reply:
Exactly 😄